src/Nethermind/Nethermind.Consensus/Processing/BlockProcessor.cs

```diff
@@ -256,10 +256,12 @@ private void ValidateProcessedBlock(Block suggestedBlock, ProcessingOptions opti
         ReceiptsTracer.SetOtherTracer(blockTracer);
         ReceiptsTracer.StartNewBlockTrace(block);
 
-        _beaconBlockRootHandler.ApplyContractStateChanges(block, spec, _stateProvider);
-        _blockhashStore.ApplyBlockhashStateChanges(block.Header);
+        ITxTracer stateTracer = blockTracer.StartNewTxTrace(null);
+        _beaconBlockRootHandler.ApplyContractStateChanges(block, spec, _stateProvider, stateTracer);
+        _blockhashStore.ApplyBlockhashStateChanges(block.Header, stateTracer);
 
-        _stateProvider.Commit(spec, commitStorageRoots: false);
+        _stateProvider.Commit(spec, stateTracer, commitStorageRoots: false);
+        blockTracer.EndTxTrace();
```

Comment:
this needs to be conditional, either a derived BlockProcessor, or some condition provided by ITxTracer